Source/DSPTool/DSPTool.cpp

```diff
@@ -65,27 +70,29 @@ static void CodeToHeader(const std::vector<u16>& code, std::string filename, con
   header.append("\n\t},\n");
 
   header.append("};\n");
+  return header;
 }
 
-static void CodesToHeader(const std::vector<u16>* codes, const std::vector<std::string>* filenames,
-                          u32 num_codes, const char* name, std::string& header)
+static std::string CodesToHeader(const std::vector<std::vector<u16>>& codes,
+                                 const std::vector<std::string>* filenames)
 {
   std::vector<std::vector<u16>> codes_padded;
-  u32 reserveSize = 0;
-  for (u32 i = 0; i < num_codes; i++)
+  std::size_t reserve_size = 0;
+  for (std::size_t i = 0; i < codes.size(); i++)
   {
     codes_padded.push_back(codes[i]);
     // Pad with nops to 32byte boundary
-    while (codes_padded.at(i).size() & 0x7f)
-      codes_padded.at(i).push_back(0);
+    while (codes_padded[i].size() & 0x7f)
+      codes_padded[i].push_back(0);
 
-    reserveSize += (u32)codes_padded.at(i).size();
+    reserve_size += codes_padded[i].size();
   }
-  header.clear();
-  header.reserve(reserveSize * 4);
-  header.append(StringFromFormat("#define NUM_UCODES %u\n\n", num_codes));
+
+  std::string header;
+  header.reserve(reserve_size * 4);
+  header.append(StringFromFormat("#define NUM_UCODES %zu\n\n", codes.size()));
   header.append("const char* UCODE_NAMES[NUM_UCODES] = {\n");
-  for (u32 i = 0; i < num_codes; i++)
+  for (std::size_t i = 0; i < codes.size(); i++)
```

Comment:
This feels a little odd, using `codes.size()` as bounds but never accessing `codes` inside the loop.
Also, range-based for loop perhaps?

Reply:
Indeed, despite having the same size, this should be using `filenames` instead.